### PR TITLE
Fix: Fail gracefully when previewing a doc with deleted reps

### PR DESCRIPTION
**Pull request by jeremypress**
todo
- [x] tests

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jeremypress has signed the CLA. Thanks for the pull request!

File: src/lib/viewers/doc/DocBaseViewer.js [hunk 1]
```diff
@@ -662,8 +662,19 @@ class DocBaseViewer extends BaseViewer {
                 // eslint-disable-next-line
                 console.error(err);
 
+                // pdf.js gives us the status code in their error message
+                const { status } = err;
+
                 // Display a generic error message but log the real one
-                const error = new PreviewError(ERROR_CODE.CONTENT_DOWNLOAD, __('error_document'), {}, err.message);
+                const error =
+                    status === 202
```
Comment (jstoffan) [inline src/lib/viewers/doc/DocBaseViewer.js]: Does pdf.js recognize 202 as an error code or is this placeholder?

File: src/lib/viewers/doc/DocBaseViewer.js [hunk 2]
```diff
@@ -662,8 +662,19 @@ class DocBaseViewer extends BaseViewer {
...
+                        ? new PreviewError(
+                            ERROR_CODE.DELETED_REPS,
+                            __('error_refresh'),
+                            { isRepDeleted: true },
+                            err.message
```
Comment (jstoffan) [inline src/lib/viewers/doc/DocBaseViewer.js]: Can we destructure `message`, as well?

File: src/lib/viewers/doc/DocBaseViewer.js (hunk 1 above)
Comment (jeremypress) [reply src/lib/viewers/doc/DocBaseViewer.js]: it's recognized as a success, but then PDF.js will throw a `MissingPDFException`, which includes the HTTP status code that led to the error.

File: test/integration/sanity/DeletedReps.e2e.test.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        cy.server();
+        // Make the representation content call to always return a 202 with no body
+
+        // Mocking requests for non original reps
+        cy.route({
+            method: 'GET',
+            url: '**/internal_files/**',
+            status: 202,
+            response: {}
+        }).as('bar');
```
Comment (ConradJChan) [inline test/integration/sanity/DeletedReps.e2e.test.js]: Are these aliases used elsewhere?

File: test/integration/sanity/DeletedReps.e2e.test.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            onBeforeLoad (win) {
+                // Workaround for fetch detection in cypress mocking. https://github.com/cypress-io/cypress/issues/95
+                delete win.fetch //eslint-disable-line
+            }
+        })
+    });
+
+    describe('Document Viewers', () => {
+        it('Should correctly handle a document with deleted reps', () => {
+            helpers.checkDeletedRepError();
```
Comment (ConradJChan) [inline test/integration/sanity/DeletedReps.e2e.test.js]: How does this guarantee that the assertion on error happens?

Comment (jeremypress) [reply test/integration/sanity/DeletedReps.e2e.test.js]: if this assertion fails, an uncaught exception will be thrown and the test will time out.

File: test/integration/sanity/DeletedReps.e2e.test.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            
+            cy.showPreview(token, fileIdDoc, { showAnnotations: false });
+
+        });
+
+        it('Should correctly handle a presentation with deleted reps', () => {
+            helpers.checkDeletedRepError();
+
+            cy.showPreview(token, presentationIdDoc, { showAnnotations: false });
```
Comment (jstoffan) [inline test/integration/sanity/DeletedReps.e2e.test.js]: Can you add a comment as to why annotations is disabled.

**Pull request merged**